### PR TITLE
Add the display.is_displaying_page condition

### PR DESCRIPTION
**Pull request by numo68**
# What does this implement/fix? 

I have added
-  `const DisplayPage *get_active_page() const` to the `DisplayBuffer`
- a condition `display.is_displaying_page` with the `id` (display) and `page_id` (page) configuration, returning true if the specified page is currently displayed at the specified display

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (this will require users to update their yaml configuration files to keep working)

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1079
 
None yet, as I want to hear whether this is the correct way to do first.

# Test Environment

- [x] ESP32
- [ ] ESP8266
- [ ] Windows
- [ ] Mac OS
- [ ] Linux

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
display:
  - platform: ili9341
    model: TFT 2.4
    id: 'tft'
    rotation: 180
    cs_pin: 22
    dc_pin: 21
    led_pin: 5

    pages:
      - id: page_1
        lambda: |-
          it.fill(COLOR_OFF);
          it.circle(120, 80, 10, COLOR_ON);
      - id: page_2
        lambda: |-
          it.fill(COLOR_OFF);
          it.circle(120, 240, 10, COLOR_ON);
  
interval:
  - interval: 2s
    then:
      - if:
          condition:
            display.is_displaying_page: page_1
          then:
            - display.page.show: page_2
          else:
            - if:
                condition:
                  display.is_displaying_page:
                    id: tft
                    page_id: page_2
                then:
                  - display.page.show: page_1
```

# Explain your changes

In order to develop a touch screen controlled gui (see #1542) using multiple pages on a tft display I needed a way to query whether a specific page is visible on the display so I can route the touch screen virtual buttons to the gui element I want to control.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

Thanks

### CODE REVIEW

**Comment by numo68** [comment]
@glmnet based on the discussion in the xpt2046 I created this condition. I would be thankful for a review or comments, thanks.

File: esphome/components/display/__init__.py [hunk 1]
```diff
@@ -125,6 +128,25 @@ def display_page_show_previous_to_code(config, action_id, template_arg, args):
     yield cg.new_Pvariable(action_id, template_arg, paren)
 
 
+@automation.register_condition(
+    "display.is_displaying_page",
+    DisplayIsDisplayingPageCondition,
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.use_id(DisplayBuffer),
```
Comment (glmnet) [inline esphome/components/display/__init__.py:136]: Check `dfplayer.play`, if there is only one display in use (most likely) you can create yaml like:


`display.is_displaying_page: page1`

**Review by glmnet — COMMENTED** [review]
Looks good, also add docs

File: esphome/components/display/__init__.py (hunk 1 above)
Comment (numo68) [reply esphome/components/display/__init__.py]: done

**Comment by numo68** [comment]
I have added the possibility of the simplified form ``display.is_displaying_page: page_1``  and created a PR for the docs

**Review by glmnet — APPROVED** [review]

**Comment by glmnet** [comment]
LGTM, docs too

**Pull request merged**